src/container-type-context.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import {useScreenSize} from './hooks';
+
+import type {ContainerType} from './utils/types';
+
+const ContainerTypeContext = React.createContext<ContainerType | null>(null);
```

Comment:
I'd call it `ContainerWidthContext` with values `'large' | 'medium' | 'small'` and only listen for them in desktop viewports. I won't mix screen size with container width

Reply:
And what about the grid layout when its displayed in a mobile or tablet ? In these cases, the expected behavior is to be displayed as if the grid won't exists.